lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -917,6 +915,60 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
     }
     return hasPrintedAnything;
   }
+    
+  void printRenameForDecl(const AvailableAttr *AvAttr, const Decl *D,
+                          bool includeQuotes) {
+    if (AvAttr->Rename.empty() && isa<ValueDecl>(D))
+      return ;
+    
+    auto renamedParsedDeclName = parseDeclName(AvAttr->Rename);
+    auto renamedDeclName = renamedParsedDeclName.formDeclName(D->getASTContext());
+    
+    auto declContext = D->getDeclContext();
+    ValueDecl *renamedFuncDecl = nullptr;
+    
+    if (isa<ClassDecl>(D) || isa<ProtocolDecl>(D)) {
+      UnqualifiedLookup lookup(renamedDeclName.getBaseIdentifier(),
+                               declContext->getModuleScopeContext(),
+                               nullptr,
+                               SourceLoc(),
+                               UnqualifiedLookup::Flags::TypeLookup);
+      renamedFuncDecl = lookup.getSingleTypeResult();
+    } else {
+      SmallVector<ValueDecl *, 4> lookupResults;
+      declContext->lookupQualified(declContext->getSelfTypeInContext(),
+                                   renamedDeclName, NL_QualifiedDefault, NULL,
+                                   lookupResults);
+      for (auto candidate : lookupResults) {
+        if (!shouldInclude(candidate))
+          continue;
+        
+        if (candidate->getKind() != D->getKind() ||
+            (candidate->isInstanceMember() !=
+             cast<ValueDecl>(D)->isInstanceMember()))
+          continue;
+        
+        if (isa<FuncDecl>(candidate) &&
+            (cast<FuncDecl>(candidate)->getParameters()->size() !=
+             cast<FuncDecl>(D)->getParameters()->size()))
+          continue;
+        
+        if (renamedFuncDecl)
+        /* TODO: Diagnose to compiler to tell the user that we found multiple candidates */;
```

Comment:
Still gotta do *something* here before we check this in. Just keeping track of a second boolean flag is probably fine.

Reply:
So we should just fail here and fallback to the `AvAttr->Rename` here?

Reply:
Yeah, the same thing we'd do for a "we found nothing that matches" case.